lib/lumberjack/beats/server.rb

```diff
@@ -66,8 +66,16 @@ def run(&block)
         # unless the server is closing
         next unless connection
 
+        Thread.abort_on_exception = true
         Thread.new(connection) do |connection|
-          connection.run(&block)
+          begin
+            connection.run(&block)
+          rescue Lumberjack::Beats::Connection::ConnectionClosed 
```

Comment:
Correct, anything that can terminate the connection will raise this exception and wrap the original exception if further action are needed.